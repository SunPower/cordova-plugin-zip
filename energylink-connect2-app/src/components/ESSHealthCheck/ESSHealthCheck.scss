@import '~@sunpower/theme-dark/styles/variables';

.ess-hc {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .hc-missing-report {
    border: 1px solid $silver-sand3;
    border-radius: 5px;
    align-items: center;
    justify-content: center;
  }

  .status-message {
    display: flex;
    flex-direction: column;
  }

  .discovery-percentage {
    line-height: 3rem;
  }

  &.gridit {
    display: grid;
    grid-template-rows: minmax(15px, 40px) 1fr;
    align-items: center;
    height: 100%;
  }

  .buttons {
    display: grid;
    grid-template-columns: 1fr 1fr;
    button {
      max-width: 120px;
    }
  }

  .collapsible.mb-10 {
    position: relative;
    overflow-x: scroll;
    .sp-hey,
    .sp-check {
      position: absolute;
      right: 12px;
      top: 12px;
    }
  }

  .check-in-progress {
    flex-direction: column;
  }
}
