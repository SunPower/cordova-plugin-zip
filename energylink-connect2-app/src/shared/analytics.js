import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
