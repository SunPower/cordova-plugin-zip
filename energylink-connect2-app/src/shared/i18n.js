import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
