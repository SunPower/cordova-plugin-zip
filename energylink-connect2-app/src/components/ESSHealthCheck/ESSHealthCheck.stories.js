import React from 'react'
import { storiesOf } from '@storybook/react'
import { linkTo } from '@storybook/addon-links'

import ESSHealthCheck from '.'
import { action } from '@storybook/addon-actions/dist'

const results = {
  errors: [],
  ess_report: {
    battery_status: [
      {
        battery_amperage: {
          unit: 'A',
          value: 0
        },
        battery_voltage: {
          unit: 'V',
          value: 53.2
        },
        last_updated: '2020-07-14 23:03:59',
        serial_number: '',
        state_of_charge: {
          unit: '%',
          value: 0.88
        },
        temperature: {
          unit: 'C',
          value: 26.3
        }
      }
    ],
    ess_state: [
      {
        operational_mode: 'STANDBY',
        permission_to_operate: false,
        storage_controller_status: 'NOT_RUNNING'
      }
    ],
    ess_status: [
      {
        enclosure_humidity: {
          unit: '%',
          value: 21
        },
        enclosure_temperature: {
          unit: 'C',
          value: 33
        },
        ess_meter_reading: {
          agg_power: {
            unit: 'kW',
            value: -0.028
          },
          last_updated: '2020-07-14 23:03:59',
          meter_a: {
            reading: {
              current: {
                unit: 'A',
                value: 0
              },
              last_updated: '2020-07-14 23:03:59',
              power: {
                unit: 'W',
                value: 0
              },
              voltage: {
                unit: 'V',
                value: 120.26
              }
            }
          },
          meter_b: {
            reading: {
              current: {
                unit: 'A',
                value: 0
              },
              last_updated: '2020-07-14 23:03:59',
              power: {
                unit: 'W',
                value: 0
              },
              voltage: {
                unit: 'V',
                value: 120.26
              }
            }
          }
        },
        last_updated: '2020-07-14 23:03:59',
        serial_number: '00001B3E5A44_29_17'
      }
    ],
    hub_plus_status: {
      aux_port_voltage: {
        unit: 'V',
        value: 11.364
      },
      contactor_error: 'NONE',
      contactor_position: 'CLOSED',
      grid_frequency_state: 'METER_FREQ_IN_RANGE',
      grid_phase1_voltage: {
        unit: 'V',
        value: 121.2
      },
      grid_phase2_voltage: {
        unit: 'V',
        value: 121.30000000000001
      },
      grid_voltage_state: 'METER_VOLTAGE_IN_RANGE',
      hub_humidity: {
        unit: '%',
        value: 26
      },
      hub_temperature: {
        unit: 'C',
        value: 29
      },
      inverter_connection_voltage: {
        unit: 'V',
        value: 0.267
      },
      jump_start_voltage: {
        unit: 'V',
        value: 0.748
      },
      last_updated: '2020-07-14 23:04:07',
      load_frequency_state: 'METER_FREQ_IN_RANGE',
      load_phase1_voltage: {
        unit: 'V',
        value: 121.2
      },
      load_phase2_voltage: {
        unit: 'V',
        value: 121.4
      },
      load_voltage_state: 'METER_VOLTAGE_IN_RANGE',
      main_voltage: {
        unit: 'V',
        value: 11.243
      },
      serial_number: 'Serial-MIDC-Mercury'
    },
    inverter_status: [
      {
        a_n_voltage: {
          unit: 'V',
          value: 120.26
        },
        ac_current: {
          unit: 'A',
          value: 0
        },
        ac_power: {
          unit: 'kW',
          value: -0.028
        },
        b_n_voltage: {
          unit: 'V',
          value: 120.26
        },
        last_updated: '2020-07-14 23:03:59',
        phase_a_current: {
          unit: 'A',
          value: 0
        },
        phase_b_current: {
          unit: 'A',
          value: 0
        },
        serial_number: '00001B3E5A44',
        temperature: {
          unit: 'C',
          value: 27.68
        }
      }
    ],
    last_updated: '2020-07-14 23:04:09'
  }
}

const actions = {
  onRetry: linkTo('ESSHealthCheck Page', 'Generating report'),
  onContinue: action('Continue action triggered'),
  onSeeErrors: linkTo('ESSHealthCheck Error List')
}

storiesOf('ESSHealthCheck Page', module)
  .add('Generating report', () => (
    <div className="full-min-height pl-10 pr-10">
      <ESSHealthCheck {...actions} />
    </div>
  ))
  .add('Error fetching information', () => (
    <div className="full-min-height pl-10 pr-10">
      <ESSHealthCheck error="UNABLE_TO_FETCH" {...actions} />
    </div>
  ))
  .add('When errors and results found', () => (
    <div className="full-min-height pl-10 pr-10">
      <ESSHealthCheck results={results} {...actions} />
    </div>
  ))
  .add('When results and no errors found', () => (
    <div className="full-min-height pl-10 pr-10">
      <ESSHealthCheck
        results={{ ess_report: results.ess_report }}
        {...actions}
      />
    </div>
  ))
