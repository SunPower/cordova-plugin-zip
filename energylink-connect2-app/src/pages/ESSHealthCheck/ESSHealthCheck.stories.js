import React from 'react'
import { storiesOf } from '@storybook/react'
import { configureStore } from 'state/store'
import { Provider } from 'react-redux'

import ESSHealthCheck from './index'

storiesOf('Storage - Health Check', module).add('Faulty Report', () => {
  const { store } = configureStore({
    storage: {
      status: {
        error: '',
        waiting: false,
        results: {
          errors: [],
          ess_report: {
            battery_status: [null],
            ess_state: [
              {
                operational_mode: 'SELF_CONSUMPTION',
                permission_to_operate: false,
                storage_controller_status: 'RUNNING'
              }
            ],
            ess_status: [
              {
                enclosure_humidity: {
                  unit: '%',
                  value: 17
                },
                enclosure_temperature: {
                  unit: 'C',
                  value: 30
                },
                ess_meter_reading: null,
                last_updated: '2020-09-29 23:54:43',
                serial_number: '00001B3DDACD_178_15'
              }
            ],
            hub_plus_status: {
              aux_port_voltage: {
                unit: 'V',
                value: 11.353
              },
              contactor_error: 'NONE',
              contactor_position: 'CLOSED',
              grid_frequency_state: 'METER_FREQ_IN_RANGE',
              grid_phase1_voltage: {
                unit: 'V',
                value: 123
              },
              grid_phase2_voltage: {
                unit: 'V',
                value: 122.60000000000001
              },
              grid_voltage_state: 'METER_VOLTAGE_IN_RANGE',
              hub_humidity: {
                unit: '%',
                value: 21
              },
              hub_temperature: {
                unit: 'C',
                value: 28
              },
              inverter_connection_voltage: {
                unit: 'V',
                value: 0.267
              },
              jump_start_voltage: {
                unit: 'V',
                value: 0.594
              },
              last_updated: '2020-09-29 23:54:51',
              load_frequency_state: 'METER_FREQ_IN_RANGE',
              load_phase1_voltage: {
                unit: 'V',
                value: 122.80000000000001
              },
              load_phase2_voltage: {
                unit: 'V',
                value: 122.60000000000001
              },
              load_voltage_state: 'METER_VOLTAGE_IN_RANGE',
              main_voltage: {
                unit: 'V',
                value: 11.254
              },
              serial_number: 'Serial-MIDC-Saturn'
            },
            inverter_status: [null],
            last_updated: '2020-09-29 23:54:53'
          }
        }
      }
    },
    devices: {
      progress: 100
    },
    rma: {
      pvs: true
    }
  })

  return (
    <div className="full-min-height pl-10 pr-10">
      <Provider store={store}>
        <ESSHealthCheck />
      </Provider>
    </div>
  )
})
