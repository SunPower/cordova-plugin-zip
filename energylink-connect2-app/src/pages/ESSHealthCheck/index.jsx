import {
  add,
  isEmpty,
  isNil,
  length,
  path,
  pathOr,
  pluck,
  propOr,
  reduce
} from 'ramda'
import React, { useEffect, useRef, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'

import ESSHealthCheckComponent from 'components/ESSHealthCheck'
import paths from 'routes/paths'
import { addHasErrorProp, warningsLength, withoutInfoCodes } from 'shared/utils'
import { RESET_DISCOVERY } from 'state/actions/devices'
import { GET_ESS_STATUS_INIT, RUN_EQS_SYSTEMCHECK } from 'state/actions/storage'
import {
  SUBMIT_CLEAR,
  ALLOW_COMMISSIONING
} from 'state/actions/systemConfiguration'

function ESSHealthCheck() {
  const dispatch = useDispatch()
  const history = useHistory()
  const unblockHandle = useRef()

  const [waitModal, showWaitModal] = useState(false)
  const [modelsWarning, toggleModelsWarning] = useState(false)
  const { rmaMode } = useSelector(state => state.rma)
  const { waiting, results, error } = useSelector(state => state.storage.status)
  const { found, progress } = useSelector(state => state.devices)
  const { canCommission } = useSelector(path(['systemConfiguration', 'submit']))
  const rmaPvs = useSelector(path(['rma', 'pvs']))

  const discoveryProgress = propOr([], 'progress', progress)
  const deviceProgress = pluck('PROGR', discoveryProgress)
  const deviceStartProgress =
    waiting && deviceProgress === 100 ? 0 : deviceProgress
  const overallProgress = Math.floor(
    reduce(add, 0, deviceStartProgress) / length(deviceProgress)
  )
  const { submitting, commissioned, error: syncError } = useSelector(
    pathOr({}, ['systemConfiguration', 'submit'])
  )

  const report = addHasErrorProp(results)
  const errors = pathOr([], ['errors'], report)

  const noInfo = withoutInfoCodes(errors)
  const warningsCount = warningsLength(noInfo)

  useEffect(() => {
    unblockHandle.current = history.block(() => {
      if (waiting) {
        showWaitModal(true)
        return false
      }
    })
    return function() {
      unblockHandle.current && unblockHandle.current()
    }
  }, [unblockHandle, history, showWaitModal, waiting])

  useEffect(() => {
    if (
      results &&
      warningsCount === length(noInfo) &&
      canCommission === false
    ) {
      dispatch(ALLOW_COMMISSIONING())
    }
  }, [canCommission, dispatch, noInfo, results, warningsCount])

  useEffect(() => {
    if (isEmpty(results) || isNil(results)) {
      dispatch(RESET_DISCOVERY({ keepDeviceList: true }))
      dispatch(GET_ESS_STATUS_INIT())
    }
  }, [dispatch, report, results])

  const onRetry = () =>
    discoveryProgress === 100
      ? dispatch(RUN_EQS_SYSTEMCHECK())
      : dispatch(GET_ESS_STATUS_INIT())

  const pathToContinue = rmaPvs
    ? paths.PROTECTED.SYSTEM_CONFIGURATION.path
    : paths.PROTECTED.INSTALL_SUCCESS.path

  const pathToErrors = paths.PROTECTED.ESS_HEALTH_CHECK_ERRORS.path

  const clearAndContinue = () => {
    dispatch(SUBMIT_CLEAR())
    history.push(paths.PROTECTED.SYSTEM_CONFIGURATION.path)
  }

  const validateModels = () => {
    const filterModels = found.filter(
      mi => mi.DEVICE_TYPE === 'Inverter' && !mi.PANEL
    )
    return filterModels.length < 1
  }

  const warnMissingModels = () => {
    if (validateModels()) {
      history.push(pathToContinue)
    } else {
      toggleModelsWarning(true)
    }
  }

  return (
    <ESSHealthCheckComponent
      waiting={waiting}
      progress={overallProgress}
      results={results}
      error={error}
      onRetry={onRetry}
      pathToContinue={pathToContinue}
      pathToErrors={pathToErrors}
      rmaMode={rmaMode}
      clear={clearAndContinue}
      submitting={submitting}
      commissioned={commissioned}
      syncError={syncError}
      waitModal={waitModal}
      showWaitModal={showWaitModal}
      modelsWarning={modelsWarning}
      toggleModelsWarning={toggleModelsWarning}
      warnMissingModels={warnMissingModels}
    />
  )
}

export default ESSHealthCheck
