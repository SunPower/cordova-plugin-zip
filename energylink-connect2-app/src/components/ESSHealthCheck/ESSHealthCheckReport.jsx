import React from 'react'
import { map, keys, propOr } from 'ramda'
import moment from 'moment'
import clsx from 'clsx'
import { useI18n } from 'shared/i18n'

function ESSHealthCheckReport({ report }) {
  const t = useI18n()

  return (
    <>
      <section>
        <h6 className="has-text-white has-text-weight-bold mb-10 mt-10">
          {t('BATTERIES')}
        </h6>
        {map(renderItem, report.battery_status)}
      </section>

      <section>
        <h6 className="has-text-white has-text-weight-bold mb-10">
          {t('INVERTERS')}
        </h6>
        {map(renderItem, report.inverter_status)}
      </section>

      <section>
        <h6 className="has-text-white has-text-weight-bold mb-10">
          {t('HUB_PLUS')}
        </h6>
        {renderItem(report.hub_plus_status)}
      </section>

      <section>
        <h6 className="has-text-white has-text-weight-bold mb-10">
          {t('ALL_IN_ONE')}
        </h6>
        {map(renderItem, report.ess_status)}
      </section>
    </>
  )
}

const renderItem = item => <ESSHealthCheckReportItem {...item} />

function ESSHealthCheckReportItem({
  serial_number,
  last_updated,
  hasError,
  ...rest
}) {
  const t = useI18n()
  const icon = clsx('is-size-5', {
    'sp-hey has-text-primary': hasError,
    'sp-check has-text-white': !hasError
  })

  return (
    <div className="collapsible mb-10">
      <i className={icon} />
      <p className="has-text-white has-text-weight-bold mb-10">
        {serial_number}
      </p>
      <p>
        <span className="has-text-weight-bold mr-5">{t('LAST_UPDATED')}:</span>
        {moment(last_updated).format('MM/DD/YYYY hh:mm A')}
      </p>

      {map(renderRestValues(t, rest), keys(rest))}
    </div>
  )
}

const renderRestValues = (t, rest) => key => (
  <p>
    <span className="mr-5 has-text-weight-bold">{t(key)}:</span>
    <span className="mr-5">{propOr(rest[key], 'value', rest[key])}</span>
    {propOr('', 'unit', rest[key])}
  </p>
)

export default ESSHealthCheckReport
