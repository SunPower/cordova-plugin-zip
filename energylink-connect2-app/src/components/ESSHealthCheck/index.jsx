import React from 'react'
import clsx from 'clsx'
import { isEmpty, length, pathOr, isNil, path } from 'ramda'
import { useI18n } from 'shared/i18n'
import { either, addHasErrorProp, warningsLength } from 'shared/utils'
import { rmaModes } from 'state/reducers/rma'

import ESSHealthCheckReport from './ESSHealthCheckReport'
import ErrorDetected from 'components/ESSErrorDetected'
import ContinueFooter from 'components/ESSContinueFooter'
import StorageSyncFooter from 'components/ESSContinueFooter/StorageSyncFooter'

import './ESSHealthCheck.scss'
import SwipeableBottomSheet from 'react-swipeable-bottom-sheet'

function ESSHealthCheck(props) {
  const t = useI18n()
  const results = addHasErrorProp(props.results)
  const errors = pathOr([], ['errors'], results)
  const report = path(['ess_report'], results)
  const loading = isNil(results) && !props.error
  const classes = clsx('ess-hc has-text-centered pt-10 pl-10 pr-10', {
    gridit: loading || props.error
  })
  const hasErrors = !isEmpty(errors) || props.error ? true : false

  const statusErrorMessage = pathOr(
    props.error,
    ['error', 'response', 'body', 'result'],
    props
  )

  const {
    waiting,
    progress,
    onRetry,
    pathToContinue,
    pathToErrors,
    rmaMode = rmaModes.NONE,
    sync,
    clear,
    submitting,
    commissioned,
    syncError,
    waitModal,
    showWaitModal
  } = props

  return (
    <div className={classes}>
      <span className="is-uppercase has-text-weight-bold">
        {t('HEALTH_CHECK')}
      </span>

      <div className="status-message">
        {either(
          waiting,
          <span className="discovery-percentage mb-20 has-text-weight-bold is-size-1 has-text-white">
            {progress || 0}%
          </span>
        )}
        {either(loading, <span> {t('HEALTH_REPORT')} </span>)}

        {either(
          !props.error && report,
          <ESSHealthCheckReport report={report} />
        )}
      </div>

      <SwipeableBottomSheet
        shadowTip={false}
        open={waitModal}
        onChange={() => showWaitModal(!waitModal)}
      >
        <div className="check-in-progress is-flex">
          <span className="has-text-weight-bold">{t('HOLD_ON')}</span>
          <span className="mt-10 mb-10">{t('WAIT_FOR_CHECK')}</span>
          <div className="mt-10 mb-20">
            <button
              className="button is-primary"
              onClick={() => showWaitModal(false)}
            >
              {t('CLOSE')}
            </button>
          </div>
        </div>
      </SwipeableBottomSheet>

      {either(
        !hasErrors && !loading,
        rmaMode === rmaModes.EDIT_DEVICES ? (
          <StorageSyncFooter
            submitting={submitting}
            commissioned={commissioned}
            error={syncError}
            sync={sync}
            clear={clear}
          />
        ) : (
          <ContinueFooter
            url={pathToContinue}
            text={t('HEALTH_CHECK_SUCCESSFUL')}
          />
        )
      )}

      {either(
        hasErrors && !loading,
        <ErrorDetected
          number={length(errors) - warningsLength(errors)}
          warnings={warningsLength(errors)}
          onRetry={onRetry}
          url={pathToErrors}
          globalError={statusErrorMessage}
          next={pathToContinue}
        />
      )}
    </div>
  )
}

export default ESSHealthCheck
