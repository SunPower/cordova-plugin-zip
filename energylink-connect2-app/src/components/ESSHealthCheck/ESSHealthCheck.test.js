import React from 'react'
import * as reactRedux from 'react-redux'
import ESSHealthCheck from '.'
import * as i18n from 'shared/i18n'

describe('ESSHealthCheck component', () => {
  describe('initial state', () => {
    let dispatchMock
    let initialState = {
      results: {
        errors: []
      }
    }

    beforeEach(() => {
      dispatchMock = jest.fn()
      jest
        .spyOn(reactRedux, 'useDispatch')
        .mockImplementation(() => dispatchMock)
      jest
        .spyOn(i18n, 'useI18n')
        .mockImplementation(path => (key, ...params) =>
          `${key.toUpperCase()} ${params.join('_')}`.trim()
        )
    })

    test('render correctly', () => {
      const component = mountWithProvider(<ESSHealthCheck {...initialState} />)(
        {}
      )
      expect(component).toMatchSnapshot()
    })
  })
})
