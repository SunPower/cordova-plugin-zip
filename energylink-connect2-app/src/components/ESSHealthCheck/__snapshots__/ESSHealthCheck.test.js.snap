// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESSHealthCheck component initial state render correctly 1`] = `
Object {
  "component": <BrowserRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <ESSHealthCheck
          results={
            Object {
              "errors": Array [],
            }
          }
        >
          <div
            className="ess-hc page-height has-text-centered pt-10"
          >
            <span
              className="is-uppercase has-text-weight-bold"
            >
              HEALTH_CHECK
            </span>
            <div
              className="status-messageee"
            />
            <div
              className="are-small"
            >
              <button
                className="button auto is-primary"
              >
                CONTINUE
              </button>
            </div>
          </div>
        </ESSHealthCheck>
      </Provider>
    </Router>
  </BrowserRouter>,
  "store": Object {
    "clearActions": [Function],
    "dispatch": [Function],
    "getActions": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
}
`;
